src/components/CIPipelineN/VariableDataTable/utils.tsx

```diff
@@ -0,0 +1,629 @@
+                    },
+                },
+                id: id || index,
+            }
+        },
+    )
+
+export const getVariableDataTableInitialCellError = (
+    rows: VariableDataRowType[],
+    headers: DynamicDataTableHeaderType<VariableDataKeys>[],
+) =>
+    rows.reduce((acc, curr) => {
+        if (!acc[curr.id]) {
+            acc[curr.id] = headers.reduce(
+                (headerAcc, { key }) => ({ ...headerAcc, [key]: { isValid: true, errorMessages: [] } }),
+                {},
+            )
+        }
+
+        return acc
+    }, {})
+
+export const getUploadFileConstraints = ({
+    unit,
+    allowedExtensions,
+    maxUploadSize,
+}: {
+    unit: string
+    allowedExtensions: string
+    maxUploadSize: string
+}) => {
+    const unitMultiplier = unit === 'MB' ? 1024 : 1
+    return {
+        allowedExtensions: allowedExtensions
+            .split(',')
+            .map((value) => value.trim())
+            .filter((value) => !!value),
+        maxUploadSize: maxUploadSize ? parseFloat(maxUploadSize) * unitMultiplier * 1024 : null,
+    }
+}
+
+export const convertVariableDataTableToFormData = ({
+    rows,
+    cellError,
+    type,
+    activeStageName,
+    selectedTaskIndex,
+    formData,
+    formDataErrorObj,
+    validateTask,
+    calculateLastStepDetail,
+}: Pick<
+    PipelineContext,
+    | 'activeStageName'
+    | 'selectedTaskIndex'
+    | 'formData'
+    | 'formDataErrorObj'
+    | 'validateTask'
+    | 'calculateLastStepDetail'
+> & {
+    type: PluginVariableType
+    rows: VariableDataRowType[]
+    cellError: DynamicDataTableCellErrorType<VariableDataKeys>
+}) => {
+    const updatedFormData = structuredClone(formData)
+    const updatedFormDataErrorObj = structuredClone(formDataErrorObj)
+
+    const currentStepTypeVariable =
+        updatedFormData[activeStageName].steps[selectedTaskIndex].stepType === PluginType.INLINE
+            ? 'inlineStepDetail'
+            : 'pluginRefStepDetail'
+
+    const ioVariables: VariableType[] =
+        updatedFormData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable][
+            type === PluginVariableType.INPUT ? 'inputVariables' : 'outputVariables'
+        ]
+
+    const updatedIOVariables = rows.map<VariableType>(({ data, customState, id }) => {
+        const selectedIOVariable = ioVariables?.find((ioVariable) => ioVariable.id === id)
+        const {
+            askValueAtRuntime,
+            blockCustomValue,
+            choices,
+            valColumnSelectedValue,
+            isVariableRequired,
+            variableDescription,
+            fileInfo,
+        } = customState
+
+        const variableDetail: VariableType = {
+            ...selectedIOVariable,
+            value: data.val.value,
+            format: data.format.value as VariableTypeFormat,
+            name: data.variable.value,
+            description: type === PluginVariableType.INPUT ? variableDescription : data.val.value,
```

Comment:
Why are we adding the value in description?

Reply:
in case of OutputVariables value column is being used as description